folding/base/validator.py

```diff
@@ -236,9 +245,29 @@ def stop_run_thread(self):
             self.thread.join(5)
             self.is_running = False
             bt.logging.debug("Stopped")
+    def sync_background_thread(self):
```

Comment:
I would preferably want this function to be merged with `run_in_background_thread()` but we don't use it at the moment, so I had to add extra code which we don't need